src/utils/magicsquareService.ts

```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+import { useLocation } from 'react-router';
+
+const MS_PROMO_IDENT = 'msPromo';
+
+export const useSaveToMs = async ():Promise<void> => {
+  // save to magicstore - remove after campaign @anukulpandey
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const params = {};
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [key, value] of searchParams) {
+    // @ts-ignore
+    params[key] = value;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const existingParams = JSON.parse(localStorage.getItem(MS_PROMO_IDENT) as any) || {};
+  const updatedParams = { ...existingParams, ...params };
```

Comment:
check if expected param name is present and only save that param value to localStorage (not whole object)